Adding Samtec PCIE-038-02-X-D-RA (PCIe 1x)

suzizecat opened the pull request:
Adding the PCIe 1x connector footprint from Samtec

[Footprint specification here](http://suddendocs.samtec.com/prints/pcie-xxx-02-x-d-ra-footprint.pdf)

![image](https://user-images.githubusercontent.com/1909563/43801020-3679c8ca-9a92-11e8-9d6c-f8dd2324274f.png)

Samtec provides 3D model but I don't think that they are correctly licensed to fit in KiCAD
------------

Thanks for creating a pull request to contribute to the KiCad libraries! To speed up integration of your PR, please check the following items:

- [ ] Provide a URL to a datasheet for the footprint(s) you are contributing
- [ ] An example screenshot image is very helpful 
- [ ] If there are matching symbol or 3D model pull requests, provide link(s) as appropriate
- [ ] Check the output of the Travis automated check scripts - fix any errors as required

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/KiCad/kicad-footprints?pullRequest=815) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/KiCad/kicad-footprints?pullRequest=815) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/KiCad/kicad-footprints?pullRequest=815) it.</sub>

poeschlr commented:
There is no courtyard offset. (Yes samtec gives a minimum area but we still have a definition that is larger as our minimum area.) 
The silk outline should be outside of the body (as fab represents the body it should be outside the fab line)
The fab outline is not drawn correctly. (offset in x and y direction. Does not include every part of the body)

![screenshot from 2018-08-09 11-29-25](https://user-images.githubusercontent.com/18327350/43890629-8681e29a-9bc7-11e8-84c4-a3f9c75ec314.png)
![screenshot from 2018-08-09 11-26-42](https://user-images.githubusercontent.com/18327350/43890665-9a160430-9bc7-11e8-83fc-b2ba6dfc829b.png)

Misca1234 commented:
@suzizecat 
Will you address the points above ?

suzizecat commented:
I will try if i find some time to do so... Sorry about that

suzizecat commented:
I *think* that I fixed the issues, please do not hesitate to let me know

poeschlr commented:
Move the silk reference designator to the top and the value to the bottom. (outside of the outline)
There is no need to decrease the font size of the value field.
Silk should not overlap any pads. (Not even non plated ones) Keep a clearance to pads. (0.2mm as this is a very large part)

Add a pin 1 marker on the fab layer.

The hole size and pad size must be the same for non plated through holes (NPTH)

Where did you get the part size in y direction from? (You used 19.5mm where does this value come from. The 3d model i posted above seems to suggest 18.85)
The body has not been fixed.

![samtec svg](https://user-images.githubusercontent.com/18327350/52522554-3309d400-2c87-11e9-9793-eb22a537f180.png)

poeschlr commented:
Oh and i think the footprint name is not quite right. There is no 38 pin version in the datasheets. I would assume it should be named `PCIE-036-02-X-D-RA` (36 pins per row)


---

And for reference a better dimensioned drawing generated from the samtec 3d model:

![screenshot from 2019-02-09 16-45-53](https://user-images.githubusercontent.com/18327350/52522787-436f7e00-2c8a-11e9-95b6-6fc8a180d347.png)

suzizecat commented:
Damn, I'm rusty on footprint making. Thanks to point out the problematic dimensions in your reply (hadn't understood that on the previous reply)

I found the 19.5 (19.6 actually, fixed) in the datasheet link at the beginning of the post : 
http://suddendocs.samtec.com/prints/pcie-xxx-02-x-d-ra-footprint.pdf
Editing the name and then pushing.

Thanks for your time :heart:

poeschlr commented:
I think the 19.6 is for the "keepout" area not necessarely for the part outline. Sadly the datasheet is a bit bad for this part to be honest.

poeschlr commented:
You corrected the distances in y direction wrong. (Introduced a new error and overshot the one you tried to correct.)

I am still convinced that we should go with the 3d model size for the body in y direction.

![screenshot from 2019-02-12 19-15-20](https://user-images.githubusercontent.com/18327350/52658195-c3038400-2efa-11e9-9024-c13c1e8e3e68.png)

suzizecat commented:
I fixed the overshoot. 
However, I don' see how could I reduce the size of the component, since 18.85mm will fall just on the last pin row (If I don't screw the other measurements...)

poeschlr commented:
This is how the part is. Look at the 3d model. The plastic support just covers the pins enough to give strengths. (I don't quite get where your problem is. The fab outline can be on top of pads, silk does not need to go all around. Simply leave it out where it would interfere with pads. Courtyard will likely be calculated against the pads in this case.)

suzizecat commented:
I would have used Fab to define the "overall" outline, running outside of the pads.
However I get your point (and the fact that what I mention above is simply the courtyard job).
For the Silkscreen, do I have to draw the horizontal line which will go over the pads (removing silk over pad) or can I just leave it out ?

I will update the FP this evening.

suzizecat commented:
I fixed the size with respect to the 3D model dimensions. I also fixed the "pin 1 indicator" on the fab layer which has a 0.12mm width.
The silkscreen is 0.5mm away from the fab outline (splitted near pads) and the CrtYrd is 0.5mm away from the limit of the pads (center + whole size/2 + 0.5)

So, it should be fine now.

poeschlr commented:
The distance between mounting holes and front edge is not quite right
Should be 7.55mm

![Screenshot from 2019-03-24 17-49-34](https://user-images.githubusercontent.com/18327350/54882720-42bb3180-4e5d-11e9-92bb-6c43a646b720.png)

suzizecat commented:
Fixed, that snap to lines end in nigthlies is great ! :heart:

poeschlr commented:
The snap to line ends is in all version 5.1.x releases. It is not suggested to use nightly builds for contributions to the official library as it can mean the contribution is incompatible with the current stable version.

I will take a look at this later. Maybe today.

poeschlr commented:
As feared your use of nightly builds has negatively impacted this file for us. simply saving it without any change in stable will change nearly every line in the file and make future maintenance a nightmare.

So please save the file using a stable version.
